pkg/netpol/eval/internal/k8s/adminnetpol.go

```diff
@@ -234,6 +236,124 @@ func subjectSelectsPeer(anpSubject apisv1a.AdminNetworkPolicySubject, p Peer) (b
+	}
+	intPort, err := strconv.ParseInt(port, portBase, portBits)
+	if err != nil {
+		return false, err
+	}
+	for _, anpPort := range *rulePorts {
+		if !onlyOnePortFieldsSet(anpPort) {
+			return false, errors.New(fmt.Sprintf("Error in Ports : %v", anpPort) + netpolerrors.ANPPortsError)
+		}
+		switch { // only one case fits
+		case anpPort.PortNumber != nil:
+			if strings.ToUpper(protocol) != getProtocolStr(&anpPort.PortNumber.Protocol) {
+				continue
+			}
+			if int64(anpPort.PortNumber.Port) == intPort {
+				return true, nil
+			}
+		case anpPort.NamedPort != nil:
+			podProtocol, podPort := dst.GetPeerPod().ConvertPodNamedPort(*anpPort.NamedPort)
+			if !strings.EqualFold(protocol, podProtocol) {
+				continue
+			}
+			if int64(podPort) == intPort {
+				return true, nil
+			}
+		case anpPort.PortRange != nil:
+			ruleProtocol := &anpPort.PortRange.Protocol
+			if strings.ToUpper(protocol) != getProtocolStr(ruleProtocol) {
+				continue
+			}
+			if isEmptyPortRange(anpPort.PortRange.Start, anpPort.PortRange.End) {
+				return false, nil
+			}
+			if intPort >= int64(anpPort.PortRange.Start) && intPort <= int64(anpPort.PortRange.End) {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+// checkIfEgressRuleContainsConn check if the egress rule captures the given connection, if yes returns if it is passed/allowed/denied
+func checkIfEgressRuleContainsConn(rulePeers []apisv1a.AdminNetworkPolicyEgressPeer, rulePorts *[]apisv1a.AdminNetworkPolicyPort, dst Peer,
+	action, protocol, port string, isBANPrule bool) (res int, captured bool, err error) {
+	if len(rulePeers) == 0 {
+		return 0, false, errors.New(netpolerrors.ANPEgressRulePeersErr)
+	}
+	peerSelected, err := egressRuleSelectsPeer(rulePeers, dst)
+	if err != nil {
+		return 0, false, err
+	}
+	if !peerSelected {
+		return 0, false, nil
+	}
+	connSelected, err := anpPortContains(rulePorts, protocol, port, dst)
+	if err != nil {
+		return 0, false, err
+	}
+	if !connSelected {
+		return 0, false, nil
+	}
+	// if the protocol and port are in the rulePorts, then action determines what to return
+	return determineConnResByAction(action, isBANPrule)
+}
+
+// checkIfIngressRuleContainsConn check if the ingress rule captures the given connection, if yes returns if it is passed/allowed/denied
+func checkIfIngressRuleContainsConn(rulePeers []apisv1a.AdminNetworkPolicyIngressPeer, rulePorts *[]apisv1a.AdminNetworkPolicyPort,
+	src, dst Peer, action, protocol, port string, isBANPrule bool) (res int, captured bool, err error) {
+	if len(rulePeers) == 0 {
+		return 0, false, errors.New(netpolerrors.ANPIngressRulePeersErr)
+	}
+	peerSelected, err := ingressRuleSelectsPeer(rulePeers, src)
+	if err != nil {
+		return 0, false, err
+	}
+	if !peerSelected {
+		return 0, false, nil
+	}
+	connSelected, err := anpPortContains(rulePorts, protocol, port, dst)
+	if err != nil {
+		return 0, false, err
+	}
+	if !connSelected {
+		return 0, false, nil
```

Comment:
same here.. it is not clear what "-1" or "0" means..

Reply:
changed